src/rtext.c

```diff
@@ -1344,40 +1344,44 @@ Rectangle GetGlyphAtlasRec(Font font, int codepoint)
 // Get text length in bytes, check for \0 character
 unsigned int TextLength(const char *text)
 {
-    unsigned int length = 0; //strlen(text)
-
-    if (text != NULL)
-    {
-        while (*text++) length++;
-    }
+    if (!text)
+        return 0;
 
+    // use strlen since it uses vector operations
+    unsigned int length = strlen(text);
+    
     return length;
 }
 
 // Formatting of text with variables to 'embed'
-// WARNING: String returned will expire after this function is called MAX_TEXTFORMAT_BUFFERS times
-const char *TextFormat(const char *text, ...)
+const char *TextFormat(const char *format, ...)
 {
-#ifndef MAX_TEXTFORMAT_BUFFERS
-    #define MAX_TEXTFORMAT_BUFFERS 4        // Maximum number of static buffers for text formatting
-#endif
+    va_list args;
+    va_start(args, format);
 
-    // We create an array of buffers so strings don't expire until MAX_TEXTFORMAT_BUFFERS invocations
-    static char buffers[MAX_TEXTFORMAT_BUFFERS][MAX_TEXT_BUFFER_LENGTH] = { 0 };
-    static int index = 0;
+    int length = vsnprintf(NULL, 0, format, args); // Determine the length of the formatted string
+    if (length < 0)
+    {
+        // Handle error
+        va_end(args);
+        return NULL;
+    }
 
-    char *currentBuffer = buffers[index];
-    memset(currentBuffer, 0, MAX_TEXT_BUFFER_LENGTH);   // Clear buffer before using
+    char *result = (char *)malloc(length + 1); // Allocate memory for the formatted string (+1 for null terminator)
```

Comment:
I think there are custom memory allocators ([raylib.h#L119-L132](https://github.com/raysan5/raylib/blob/master/src/raylib.h#L119-L132)) (e.g.: `RL_MALLOC`). Also, just asking, doesn't this need to be freed?